### ISSUE TITLE
[BUG] Localeswitcher not correctly linking to translated slug

### ISSUE THREAD

**Issue by danielschleindlsperger** [AnimalDesign/bolt-translate#61]
Hey,

when I use the localeswitcher with translated slugs, it just changes the locale slug in the url, but not the slug of the page.

e.g. when i have a page http://example.com/en/about-us it links to http://example.com/de/about-us even though the translated german slug is "ueber-uns".

### menu.yml
```yml
main:
    - path: pages/2
    - path: pages/2
    - path: pages/2
    - path: pages/2
    - path: pages/2
```

### contenttypes.yml excerp
```yml
pages:
    name: Pages
    singular_name: Page
    fields:
        locale:
            type: locale
            group: content
        slug:
            type: slug
            uses: title
            is_translateable: true
        title:
            type: text
            class: large
            group: content
            is_translateable: true
```

### routing overwrite (problem exists without it aswell)
```yml
pagebinding:
  path:           /{slug}
  defaults:       { _controller: 'Bolt\Controllers\Frontend::record', 'contenttypeslug': 'page' }
  contenttype:    pages
```

### Localeswitcher template   
(how do i get the active locale first without looping twice? :D)
```twig
<ul{% if classes %} class="{{ classes }}"{% endif %}>

    {% for locale in locales %}
        {% if locale.get('active') %}
            <li class="active">
                <a title="{{ locale.get('label') }}" href="{{ locale.get('url') }}">
                    {{ locale.get('label') }}
                </a>
            </li>
        {% endif %}
    {% endfor %}

    {% for locale in locales %}
        {% if not locale.get('active') %}
            <li>
                <a title="{{ locale.get('label') }}" href="{{ locale.get('url') }}">
                    {{ locale.get('label') }}
                </a>
            </li>
        {% endif %}
    {% endfor %}
    
</ul>
```

 - Translate Version: 3.1.1
 - Install type: Extension Repository
 - Bolt Version: 3.0.12
 - PHP version: 7.0
 - Used webserver: Apache

**Comment by madc** [comment]
You're right, thanks for reporting. I'll have a closer look.

**Comment by SahAssar** [comment]
By the way, the "how do i get the active locale first without looping twice?" question will be solved by #63
